R/DataSetList.R

```diff
@@ -39,74 +47,91 @@ DataSetList <- function(path = NULL, verbose = T, print_fun = NULL, maximization
       print_fun(paste('Processing', file, '...\n'))
-      print_fun(sprintf('   algorithm %s...\n', indexInfo[[1]]$algId))
     }
-
-    for (info in indexInfo) {
-      if (verbose) {
-        print_fun(sprintf('      %d instances on f%d %dD...\n',
-                          length(info$instance), info$funcId, info$DIM))
+    
+    if (format == NEVERGRAD){
+      dsl <- read_nevergrad(file)
+      object %<>% c(.,dsl)
+    }
+      
+    else{
+      indexInfo <- read_IndexFile(file)
+      if (verbose){
+        print_fun(sprintf('   algorithm %s...\n', indexInfo[[1]]$algId))
       }
 
-      copy_flag <- TRUE
-      data <- DataSet(info, maximization = maximization, format = format,
-                      subsampling = subsampling)
-      DIM[i] <- attr(data, 'DIM')
-      funcId[i] <- attr(data, 'funcId')
-      algId[i] <- attr(data, 'algId')
-      instance <- attr(data, 'instance')
-
-      # check for duplicated instances
-      if (length(object) != 0) {
-        idx <- sapply(object, function(obj) obj == data) %>% which
-        for (k in idx) {
-          instance_ <- attr(object[[k]], 'instance')
-          if (all(instance == instance_)) {
-            copy_flag <- FALSE
-            warning('duplicated instances!')
-            break
-          }
-
-          if (length(intersect(instance, instance_)) != 0) {
-            warning('duplicated instances!')
+      for (info in indexInfo) {
+        if (verbose) {
+          print_fun(sprintf('      %d instances on f%d %dD...\n',
+                            length(info$instance), info$funcId, info$DIM))
+        }
+  
+        copy_flag <- TRUE
+        data <- DataSet(info, maximization = maximization, format = format,
+                        subsampling = subsampling)
+        DIM[i] <- attr(data, 'DIM')
+        funcId[i] <- attr(data, 'funcId')
+        algId[i] <- attr(data, 'algId')
+        instance <- attr(data, 'instance')
+  
+        # check for duplicated instances
+        if (length(object) != 0) {
+          idx <- sapply(object, function(obj) obj == data) %>% which
+          for (k in idx) {
+            instance_ <- attr(object[[k]], 'instance')
+            if (all(instance == instance_)) {
+              copy_flag <- FALSE
+              warning('duplicated instances!')
+              break
+            }
+  
+            if (length(intersect(instance, instance_)) != 0) {
+              warning('duplicated instances!')
+            }
           }
         }
-      }
-
-      if (copy_flag) {
-        object[[i]] <- data
-        i <- i + 1
+  
+        if (copy_flag) {
+          object[[i]] <- data
+          i <- i + 1
+        }
       }
     }
-
     if (verbose) {
       print_fun("\n")
     }
   }
 
   # TODO: sort all DataSet by multiple attributes: algId, funcId and DIM
-  class(object) %<>% c('DataSetList')
-  attr(object, 'DIM') <- DIM
-  attr(object, 'funcId') <- funcId
-  attr(object, 'algId') <- algId
+  if (format != NEVERGRAD){
```

Comment:
We might always keep those attributes, however, set them to NULL in case of nevergrad. What do you think?

Reply:
This is just an artifact of the way nevergrad works. Since a single file consists of multiple algorithms, dimensions and functions the function to read a single file (read_nevergrad) already sets these attributes, and the concatenation should then update them when reading multiple files.